compiler/hash-ast/src/ast.rs

```diff
@@ -642,12 +781,18 @@ pub struct StrLitPat(pub Str);
 pub struct CharLitPat(pub char);
 
 /// An integer literal pattern.
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct IntLitPat(pub u64);
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct IntLitPat {
+    pub value: u128,
+    pub kind: IntLitKind,
+}
 
 /// A float literal pattern.
-#[derive(Debug, PartialEq, Clone)]
-pub struct FloatLitPat(pub f64);
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct FloatLitPat {
+    pub value: f64,
+    pub kind: FloatLitKind,
+}
```

Comment:
Why does this exist?

Reply:
It exists so that it can give a concrete error to why this occurred, and it might be used in ranges as the `lo` and `hi`

Reply:
So do we support float pattern ranges? Does the literal `1.0` always match `1.0..2.0`?

Reply:
Hmm, you are right, it is probably impossible to get a sound implementation, but I don't see much point in getting rid of the node just for consistency's sake

Reply:
I guess it might be useful for errors..

Reply:
Yes, I think the error reporting could benefit from this, I don't think it adds bloat, but we could make a `Pat::Lit` in the AST which just has an inner `Literal` rather than having copies for each variant for both literals and pattern literals